core/src/test/java/uk/gov/gchq/koryphe/util/EqualityTest.java

```diff
@@ -0,0 +1,128 @@
+package uk.gov.gchq.koryphe.util;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+/**
+ * A TestClass which checks for equality and inequality using both the Equals and Hashcode methods of an object
+ * @param <T> The Class of object that is being tested against.
+ */
+public abstract class EqualityTest<T> {
```

Comment:
Would it be worth adding a shouldJsonSerialiseAndDeserialise method?

Reply:
Possibly. The reason I kept it separate is that I viewed JSON serialisation as different to equality checking. Plus each one is different. Perhaps that is something that could be addressed in a different ticket?